src/lib.rs

```diff
@@ -1465,6 +1472,40 @@ impl RelativePath {
 
     /// Constructs a relative path from the current path, to `path`.
     ///
+    /// This function will return the empty relative path `""` if the component
+    /// we are traversing *from* contains unnamed components like `..` that
+    /// would have to be traversed to reach the destination. This is necessary,
+    /// since we have no idea what the names of those components are when we're
+    /// building the new relative path.
+    ///
+    /// ```
+    /// use relative_path::RelativePath;
+    ///
+    /// // Here we don't know what directories `../..` refer to, so there's no
+    /// // way to construct a path back to `bar` from `../..`.
+    /// let r = RelativePath::new("../../foo/relative-path");
```

Comment:
It would be clearer if you name paths as base and relative, or origin and relative and stick to this naming. Currently, you use a, b, r, p, etc which is confusing.

Reply:
Went with source and destination where appropriate. Cheers!